aws_xray_sdk/ext/aiohttp/client.py

```diff
@@ -0,0 +1,53 @@
+"""
+AioHttp Client tracing, only compatible with Aiohttp 3.X versions
+"""
+import aiohttp
+import traceback
+
+from types import SimpleNamespace
+
+from aws_xray_sdk.core import xray_recorder
+from aws_xray_sdk.core.models import http
+from aws_xray_sdk.ext.util import inject_trace_header, strip_url
+
+
+async def begin_subsegment(session, trace_config_ctx, params):
+    name = trace_config_ctx.name if trace_config_ctx.name else strip_url(str(params.url))
+    subsegment = xray_recorder.begin_subsegment(name, trace_config_ctx.namespace)
+    subsegment.put_http_meta(http.METHOD, params.method)
+    subsegment.put_http_meta(http.URL, params.url.human_repr())
+    inject_trace_header(params.headers, subsegment)
+
+
+async def end_subsegment(session, trace_config_ctx, params):
+    subsegment = xray_recorder.current_subsegment()
+    subsegment.put_http_meta(http.STATUS, params.response.status)
+    xray_recorder.end_subsegment()
+
+
+async def end_subsegment_with_exception(session, trace_config_ctx, params):
+    subsegment = xray_recorder.current_subsegment()
+    subsegment.add_exception(
+        params.exception,
+        traceback.extract_stack(limit=xray_recorder._max_trace_back)
+    )
+    xray_recorder.end_subsegment()
+
+
+def aws_xray_trace_config(name=None, namespace=None):
```

Comment:
Third party library support should have non-None default value as the library has the context of what is captured. For example, `boto` defaults the namespace to `aws` and `requests` defaults the namespace to `remote`. The aiohttp client is used for making outbound calls so it should be default to `remote` with the option to be set to something else (although I cannot think of any use case that anyone needs to do that).

Setting namespace to `remote` makes sure the service graph renders a node for that endpoint. The X-Ray backend only renders a node when 1. that node actively sends segment or 2. the namespace of the subsegment representing that node is `aws` or `remote`. To be more specific, if you use the client to call a service `B` which you don't own, then `B` is unlikely to generate any trace data. If this call is marked as `remote`, you can see `B` on your service graph.

The only case where you might want to mark an out-going http call subsegment to `local` depends on the http client implementation. If the client thrown a `ClientError` when the call cannot even go out due to connection timeout, DNS lookup fails, invalid input parameters etc, you might want to flip it back to `local` as the error has nothing to do with the downstream service. This is most time a bonus as a lot of libraries doesn't let you easily categorize if the error is from local or from downstream.

Reply:
Good to know, so better remove it as a param having it as a constant using always remote?

Reply:
I would purpose to change to `def aws_xray_trace_config(name=None, namespace='remote'):`.